front/app/containers/Admin/reporting/components/ReportBuilder/TopBar/index.tsx

```diff
@@ -11,75 +11,61 @@ import { useReportContext } from 'containers/Admin/reporting/context/ReportConte
+import LocaleSwitcher from 'components/admin/ContentBuilder/TopBar/LocaleSwitcher';
 import SaveButton from 'components/admin/ContentBuilder/TopBar/SaveButton';
 import { Box, Text, Title } from '@citizenlab/cl2-component-library';
 import Modal from 'components/UI/Modal';
 import Button from 'components/UI/Button';
 import ShareReportButton from '../../ReportBuilderPage/ReportRow/Buttons/ShareReportButton';
 
+// styling
+import { colors } from 'utils/styleUtils';
+
 // i18n
 import messages from './messages';
 import { FormattedMessage } from 'utils/cl-intl';
 
-// styling
-import { fontSizes, colors, stylingConsts } from 'utils/styleUtils';
-import styled from 'styled-components';
-
 // routing
 import clHistory from 'utils/cl-router/history';
 
-// utils
-import { isNilOrError } from 'utils/helperUtils';
-import { isEqual } from 'lodash-es';
-
 // types
 import { Locale } from 'typings';
-
-import useUpdateReportLayout from 'api/report_layout/useUpdateReportLayout';
-
-const LocaleBadge = styled(Box)`
-  display: inline-block;
-  color: ${colors.textSecondary};
-  background-color: ${colors.grey200};
-  font-weight: bold;
-  font-size: ${fontSizes.xs}px;
-  padding: 0px 6px;
-  margin-left: 15px;
-  border-radius: ${stylingConsts.borderRadius};
-`;
+import { CraftJson } from 'components/admin/ContentBuilder/typings';
 
 type ContentBuilderTopBarProps = {
-  hasPendingState?: boolean;
-  localesWithError: Locale[];
-  previewEnabled: boolean;
-  setPreviewEnabled: React.Dispatch<React.SetStateAction<boolean>>;
-  selectedLocale: Locale | undefined;
-  draftEditorData?: Record<string, SerializedNodes>;
-  initialData: SerializedNodes | undefined;
+  hasError: boolean;
+  hasPendingState: boolean;
+  selectedLocale: Locale;
+  draftEditorData?: CraftJson;
   reportId: string;
   templateProjectId?: string;
+  saved: boolean;
+  previewEnabled: boolean;
+  setSaved: React.Dispatch<React.SetStateAction<boolean>>;
+  setPreviewEnabled: React.Dispatch<React.SetStateAction<boolean>>;
+  setSelectedLocale: React.Dispatch<React.SetStateAction<Locale>>;
 };
 
 const ContentBuilderTopBar = ({
-  previewEnabled,
-  setPreviewEnabled,
-  selectedLocale,
+  hasError,
   draftEditorData,
-  initialData,
-  localesWithError,
+  selectedLocale,
   hasPendingState,
   reportId,
   templateProjectId,
+  saved,
+  previewEnabled,
+  setSaved,
+  setPreviewEnabled,
+  setSelectedLocale,
 }: ContentBuilderTopBarProps) => {
   const [initialized, setInitialized] = useState(false);
   const [showQuitModal, setShowQuitModal] = useState(false);
-  const [hasChange, setHasChange] = useState(false);
   const { query } = useEditor();
-  const { data: report } = useReport(reportId);
   const { mutate: updateReportLayout, isLoading } = useUpdateReportLayout();
   const reportContext = useReportContext();
 
-  const disableSave = localesWithError.length > 0;
+  const disableSave = !!hasError || !!hasPendingState || saved;
```

Comment:
Why disable the save if it's already saved?

Reply:
Because if you already saved all your current changes, you don't need to save it again (until you have new changes)